rskj-core/src/test/java/co/rsk/vm/opcode/MCopyDslTest.java

```diff
@@ -0,0 +1,851 @@
+
+    }
+
+    @Test
+    void testMCOPY_testCase4_behavesAsExpected() throws FileNotFoundException, DslProcessorException {
+
+        DslParser parser = DslParser.fromResource("dsl/opcode/mcopy/testCopying8BytesFromOffset0toOffset1.txt");
+        World world = new World();
+        WorldDslProcessor processor = new WorldDslProcessor(world);
+        processor.processCommands(parser);
+
+        // Assertions
+
+        // There's one block (b01) containing only 1 transaction
+        Block block1 = world.getBlockByName("b01");
+        Assertions.assertNotNull(block1);
+        Assertions.assertEquals(1, block1.getTransactionsList().size());
+
+        // There's a transaction called txTestMCopy
+        Transaction txTestMCopy = world.getTransactionByName("txTestMCopy");
+        Assertions.assertNotNull(txTestMCopy);
+
+        // Transaction txTestMCopy has a transaction receipt
+        TransactionReceipt txTestMCopyReceipt = world.getTransactionReceiptByName("txTestMCopy");
+        Assertions.assertNotNull(txTestMCopyReceipt);
+
+        // Transaction txTestMCopy has been processed correctly
+        byte[] creationStatus = txTestMCopyReceipt.getStatus();
+        Assertions.assertNotNull(creationStatus);
+        Assertions.assertEquals(1, creationStatus.length);
+        Assertions.assertEquals(1, creationStatus[0]);
+
+        // There's one block (b02) containing only 1 transaction
+        Block block2 = world.getBlockByName("b02");
+        Assertions.assertNotNull(block2);
+        Assertions.assertEquals(1, block2.getTransactionsList().size());
+
+        // There's a transaction called txTestMCopyOKCall
+        Transaction txTestMCopyOKCall = world.getTransactionByName("txTestMCopyOKCall");
+        Assertions.assertNotNull(txTestMCopyOKCall);
+
+        // Transaction txTestMCopyOKCall has a transaction receipt
+        TransactionReceipt txTestMCopyOKCallReceipt = world.getTransactionReceiptByName("txTestMCopyOKCall");
+        Assertions.assertNotNull(txTestMCopyOKCallReceipt);
+
+        // Transaction txTestMCopyOKCall has been processed correctly
+        byte[] txTestMCopyOKCallCreationStatus = txTestMCopyOKCallReceipt.getStatus();
+        Assertions.assertNotNull(txTestMCopyOKCallCreationStatus);
+        Assertions.assertEquals(1, txTestMCopyOKCallCreationStatus.length);
+        Assertions.assertEquals(1, txTestMCopyOKCallCreationStatus[0]);
+
+        // Check events
+        Assertions.assertEquals(1, TransactionReceiptUtil.getEventCount(txTestMCopyOKCallReceipt, "OK", null));
+        Assertions.assertEquals(0, TransactionReceiptUtil.getEventCount(txTestMCopyOKCallReceipt, "ERROR", null));
+
+    }
+
+    // Advanced Overwrite Test Cases
+    // https://github.com/ethereum/execution-spec-tests/blob/c0065176a79f89d93f4c326186fc257ec5b8d5f1/tests/cancun/eip5656_mcopy/test_mcopy.py)
+
+    @Test
+    void testMCOPY_overwriteCases_behaveAsExpected() throws FileNotFoundException, DslProcessorException {
```

Comment:
praise:

Well done my friend, this test it's very complete and it should have been a bit tricky understand the python one and convert it to a proper DSL. 👏